art/attacks/inference/attribute_inference/black_box.py

```diff
@@ -121,7 +121,7 @@ def fit(self, x: np.ndarray) -> None:
         # train attack model
         self.attack_model.fit(x_train, y_ready)
 
-    def infer(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> np.ndarray:
+    def infer(self, x: np.ndarray, y: np.ndarray = None, **kwargs) -> np.ndarray:
```

Comment:
Would you like to make the argument `y` required? If yes, we would need to remove the `= None` too.
```suggestion
    def infer(self, x: np.ndarray, y: np.ndarray, **kwargs) -> np.ndarray:
```

Reply:
Yes, you're right of course. Fixed it in another commit (and also removed the None option from the param check).